modules/33acrossBidAdapter.js

```diff
@@ -152,6 +156,42 @@ function _createServerRequest(bidRequest, gdprConsent = {}, uspConsent, pageUrl)
   }
 }
 
+function _getSchainInfo(schain) {
+  if(_isValidSchain(schain)) {
+    return {
+      ext: { schain }
+    }
+  } else {
+    utils.logWarn('[33Across Adapter] Invalid schain info passed');
+    return null;
+  }
+}
+
+// We validate schain as follows:
+//  - Complete status must be 0|1
+//  - Nodes must be defined and non-empty array
+//  - Each node must contain all the required fields
+// The entire schain schain object is marked as invalid i.e. return false if any 
+// of the above conditions fail.
+function _isValidSchain(schain) {
+  //Check of nodes are valid
+  if(!schain.nodes || !schain.nodes.length > 0) {
+    return false;
+  }
+
+  //Check is complete status is valid
+  if(schain.complete !== 0 && schain.complete !== 1) {
+    return false;
+  }
+
+  // Check for required fields in nodes
+  const requiredFields = ['asi', 'sid', 'hp'];
+
+  return schain.nodes.every((node)=> {
+    return requiredFields.every((field)=> node[field]);
```

Comment:
does empty string and 0 fail this check?

Reply:
It should yes. The values of `asi`, `sid` should be a valid string and `hp` should be 1+.

Reply:
Here's what the validation is based on https://github.com/InteractiveAdvertisingBureau/openrtb/blob/master/supplychainobject.md#openrtb-object-supplychain

Too strict?

Reply:
i see, that make sense